From ad70c7516610cb37121b94cb7bcc3c9f4cf3d3d4 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Fri, 8 Feb 2019 17:33:16 +0100
Subject: [PATCH 0992/1051] net: mvneta: call RX descriptors helpers inline

This change help increasing performance in ingress path.

Change-Id: Ic923df2b78943b1461c4822ca54a9ce098e6c3fb
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/3748
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stefan Chulski <Stefan.Chulski@cavium.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/mvneta.c | 24 ++++++++++++------------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/drivers/net/ethernet/marvell/mvneta.c b/drivers/net/ethernet/marvell/mvneta.c
index 46001b1069aa..9231ff310bc3 100644
--- a/drivers/net/ethernet/marvell/mvneta.c
+++ b/drivers/net/ethernet/marvell/mvneta.c
@@ -769,8 +769,8 @@ static void mvneta_rxq_non_occup_desc_add(struct mvneta_port *pp,
 }
 
 /* Get number of RX descriptors occupied by received packets */
-static int mvneta_rxq_busy_desc_num_get(struct mvneta_port *pp,
-					struct mvneta_rx_queue *rxq)
+static inline int mvneta_rxq_busy_desc_num_get(struct mvneta_port *pp,
+					       struct mvneta_rx_queue *rxq)
 {
 	u32 val;
 
@@ -781,9 +781,9 @@ static int mvneta_rxq_busy_desc_num_get(struct mvneta_port *pp,
 /* Update num of rx desc called upon return from rx path or
  * from mvneta_rxq_drop_pkts().
  */
-static void mvneta_rxq_desc_num_update(struct mvneta_port *pp,
-				       struct mvneta_rx_queue *rxq,
-				       int rx_done, int rx_filled)
+static inline void mvneta_rxq_desc_num_update(struct mvneta_port *pp,
+					      struct mvneta_rx_queue *rxq,
+					      int rx_done, int rx_filled)
 {
 	u32 val;
 
@@ -1621,9 +1621,9 @@ static void mvneta_tx_done_pkts_coal_set(struct mvneta_port *pp,
 }
 
 /* Handle rx descriptor fill by setting buf_cookie and buf_phys_addr */
-static void mvneta_rx_desc_fill(struct mvneta_rx_desc *rx_desc,
-				u32 phys_addr, void *virt_addr,
-				struct mvneta_rx_queue *rxq)
+static inline void mvneta_rx_desc_fill(struct mvneta_rx_desc *rx_desc,
+				       u32 phys_addr, void *virt_addr,
+				       struct mvneta_rx_queue *rxq)
 {
 	int i;
 
@@ -1819,10 +1819,10 @@ static void mvneta_txq_done(struct mvneta_port *pp,
 
 /* Refill processing for SW buffer management */
 /* Allocate page per descriptor */
-static int mvneta_rx_refill(struct mvneta_port *pp,
-			    struct mvneta_rx_desc *rx_desc,
-			    struct mvneta_rx_queue *rxq,
-			    gfp_t gfp_mask)
+static inline int mvneta_rx_refill(struct mvneta_port *pp,
+				   struct mvneta_rx_desc *rx_desc,
+				   struct mvneta_rx_queue *rxq,
+				   gfp_t gfp_mask)
 {
 	dma_addr_t phys_addr;
 	struct page *page;
-- 
2.17.1

